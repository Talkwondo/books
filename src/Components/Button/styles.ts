import {StyleSheet} from 'react-native';
import {getWidth} from '../../Utils/dimentions';

export const styles = StyleSheet.create({
  contianer: {
    alignSelf: 'center',
    display: 'flex',
    borderRadius: 10,
    paddingHorizontal: getWidth() * 0.25,
    paddingVertical: 10,
    marginVertical: 10,
  },
  text: {
    fontSize: 30,
    color: 'white',
    textAlign: 'center',
  },
});
