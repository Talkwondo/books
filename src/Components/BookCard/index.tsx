import * as React from 'react';
import {View, Text, TouchableOpacity, Button} from 'react-native';
import {BookCardProps} from '../../Types/components.modal';
import {styles} from './styles';
//@ts-ignore
import Chevron from '../../Assets/chevron-up-arrow.svg';
//@ts-ignore
import OpenChevron from '../../Assets/chevron-down.svg';

const BookCard = ({
  book,
  edit,
  setEdit,
  deleteBook,
  navigation,
}: BookCardProps) => {
  return (
    <View style={styles.contianerBook}>
      <View style={styles.row}>
        <Text style={styles.text}>{book.name}</Text>
        <Text style={styles.text}>{book.author}</Text>
        <Text style={styles.text}>{book.purchaseDay}</Text>
        <Text style={styles.text}>{book.price}$</Text>
        <TouchableOpacity
          onPress={() => {
            edit.length && edit === book.id ? setEdit('') : setEdit(book.id);
          }}>
          {edit === book.id ? (
            <OpenChevron width={30} height={30} />
          ) : (
            <Chevron width={30} height={30} />
          )}
        </TouchableOpacity>
      </View>
      {edit === book.id && (
        <View style={styles.extention}>
          <View style={styles.row}>
            <Button
              title={'edit'}
              onPress={() => {
                navigation.navigate('Modal', book);
              }}
            />
            <Button title={'delete'} onPress={() => deleteBook(book.id)} />
          </View>
        </View>
      )}
    </View>
  );
};

export default BookCard;
