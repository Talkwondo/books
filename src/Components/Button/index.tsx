import * as React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {ButtonProps} from '../../Types/components.modal';
import {styles} from './styles';
export const Button = ({
  text,
  backgroundColor,
  callback,
  customeStyle,
  customeTextStyle,
}: ButtonProps) => {
  return (
    <TouchableOpacity
      onPress={callback}
      style={[styles.contianer, {backgroundColor}, customeStyle]}>
      <Text style={[styles.text, customeTextStyle]}>{text}</Text>
    </TouchableOpacity>
  );
};
