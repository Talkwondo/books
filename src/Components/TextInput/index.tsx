import * as React from 'react';
import {TextInput, View, Text, TouchableOpacity} from 'react-native';
import {styles} from './styles';
import {
  TextInputFieldProps,
  classTextInputProps,
} from '../../Types/components.modal';

export const TextInputField = ({
  onChange,
  value,
  title,
  number,
}: TextInputFieldProps) => {
  return (
    <View style={styles.input}>
      <Text style={styles.text}>{title}</Text>
      <TextInput
        onChangeText={onChange}
        value={value}
        onFocus={() => onChange('')}
        keyboardType={number ? 'number-pad' : 'default'}
        style={styles.textInput}
      />
    </View>
  );
};

export class ButtonTextFeild extends React.Component<classTextInputProps, {}> {
  constructor(props: classTextInputProps) {
    super(props);
  }
  render() {
    return (
      <View style={styles.input}>
        <Text style={styles.text}>{this.props.title}</Text>
        <TouchableOpacity
          onPress={this.props.callback}
          style={styles.dateStyle}>
          <Text style={styles.textInput}>{this.props.date}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
export default ButtonTextFeild;
