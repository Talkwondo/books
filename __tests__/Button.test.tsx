//@ts-ignore
import renderer from 'react-test-renderer';
import React = require('react');
import {Button} from '../src/Components/Button';
describe('Snapshot', () => {
  it('to match snapshot', () => {
    const component = renderer.create(
      <Button text={'Test'} backgroundColor={'white'} callback={() => {}} />,
    );

    const match = component.toJSON();
    expect(match).toMatchSnapshot();
  });
});
