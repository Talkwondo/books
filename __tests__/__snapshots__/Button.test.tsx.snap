// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot to match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "display": "flex",
      "marginVertical": 10,
      "opacity": 1,
      "paddingHorizontal": 187.5,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontSize": 30,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Test
  </Text>
</View>
`;
